rpcserver_test.go

```diff
@@ -0,0 +1,177 @@
+package pool
+
+import (
+	"context"
+	"encoding/hex"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/wire"
+	gomock "github.com/golang/mock/gomock"
+	"github.com/lightninglabs/pool/account"
+	"github.com/lightninglabs/pool/order"
+	"github.com/lightninglabs/pool/poolrpc"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	ctxTimeout      = 1 * time.Second
+	traderKeyStr    = "036b51e0cc2d9e5988ee4967e0ba67ef3727bb633fea21a0af58e0c9395446ba09"
+	traderKeyRaw, _ = hex.DecodeString(traderKeyStr)
+)
+
+func getAccountKey(acctKey []byte) *btcec.PublicKey {
+	res, _ := btcec.ParsePubKey(acctKey, btcec.S256())
+	return res
+}
+
+func genRenewAccountReq(accountKey []byte, absolute, relative uint32,
+	feerate uint64) *poolrpc.RenewAccountRequest {
+
+	req := &poolrpc.RenewAccountRequest{}
+
+	req.AccountKey = accountKey
+	req.FeeRateSatPerKw = feerate
+	if absolute != 0 {
+		req.AccountExpiry = &poolrpc.RenewAccountRequest_AbsoluteExpiry{
+			AbsoluteExpiry: absolute,
+		}
+	} else {
+		req.AccountExpiry = &poolrpc.RenewAccountRequest_RelativeExpiry{
+			RelativeExpiry: relative,
+		}
+	}
+
+	return req
+}
+
+var renewAccountTestCases = []struct {
+	name          string
+	getReq        func() *poolrpc.RenewAccountRequest
+	checkResponse func(*poolrpc.RenewAccountResponse) error
+	mockSetter    func(*poolrpc.RenewAccountRequest,
+		*account.MockManager, *MockMarshaler,
+	)
```

Comment:
nit: closing parenthesis of method definition should be on same line as last argument (to better distinguish from method call). Same in the actual implementation below.